pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go

```diff
@@ -150,6 +153,104 @@ func TestFindAndAddNewPods_WithRescontructedVolume(t *testing.T) {
 
 }
 
+func TestFindAndAddNewPods_WithDifferentConditions(t *testing.T) {
+	tests := []struct {
+		desc           string
+		hasAddedPods   bool
+		expectedFound  bool // Found pod is added to DSW
+		setRemoved     bool
```

Comment:
I think you can replace the 2 set parameters with a single podState parameter, that has values "terminating", "removed" and "other"

Reply:
Modified with a switch case.